src/test/java/com/rocana/configuration/TestConfigurationParser.java

```diff
@@ -304,4 +305,27 @@ public void testInvalidSyntax4() throws Exception {
     parser.parse(CharSource.wrap("{ 7: 1 }"), FlatObject.class);
   }
 
+  /**
+   * In rocana-configuration 1.x, floating point values default to a float for backwards compatibility to 1.0 which
+   * didn't have a double type. This will change in 2.0 to better match Java's type system.
+   * @throws IOException If reading from the resource fails
+   */
+  @Test
+  public void testDoubleRequiresType() throws IOException {
+    ConfigurationParser parser = new ConfigurationParser();
+
+    try (InputStream inputStream = Resources.getResource("conf/double-requires-type.conf").openStream()) {
+      FlatObject object = parser.parse(
+        new InputStreamReader(
+          inputStream
+        ),
+        FlatObject.class
+      );
+
+      Assert.fail("Expected a ConfiguraitonException for unexpected type 'Float'");
+    } catch (ConfigurationException ex) {
+      Assert.assertTrue("Unexpected message in ConfigurationException: " + ex.getMessage(), ex.getMessage().startsWith("Unexpected type 'Float' at 23:2"));
```

Comment:
I'm expecting the double to not work (that's what's on line 23). I have `float-value: 1.0` before `double-value: 1.0` to ensure that the constant `1.0` is of type float. That's why the expected message check looks for the line number. Should I add a comment to make that more clear?